code/go/0chain.net/smartcontract/storagesc/block_reward.go

```diff
@@ -1,11 +1,180 @@
 package storagesc
 
 import (
+	"fmt"
+	"math"
+	"math/rand"
+	"strconv"
+
 	cstate "0chain.net/chaincore/chain/state"
+	"0chain.net/chaincore/state"
+	"0chain.net/core/common"
+	"0chain.net/core/encryption"
+	"0chain.net/core/logging"
+	"0chain.net/core/maths"
+	"0chain.net/smartcontract/partitions"
+	"0chain.net/smartcontract/stakepool/spenum"
+	"go.uber.org/zap"
 )
 
-func (ssc *StorageSmartContract) payBlobberBlockRewards(
+func (ssc *StorageSmartContract) blobberBlockRewards(
 	balances cstate.StateContextI,
 ) (err error) {
+	logging.Logger.Info("blobberBlockRewards started",
+		zap.Int64("round", balances.GetBlock().Round),
+		zap.String("block_hash", balances.GetBlock().Hash))
+
+	var (
+		stakePools  []*stakePool
+		stakeTotals []float64
+		totalQStake float64
+		weight      []float64
+		totalWeight float64
+	)
+
+	// TODO: move all the maths constants with right name once finalized to the sc.yaml
+	const (
+		// constants for gamma
+		alpha = 1
+		A     = 1
+		B     = 1
+
+		// constants for zeta
+		I  = 1
+		K  = 1
+		mu = 1
+	)
+
+	conf, err := ssc.getConfig(balances, true)
+	if err != nil {
+		return common.NewError("blobber_block_rewards_failed",
+			"cannot get smart contract configurations: "+err.Error())
+	}
+
+	if conf.BlockReward.BlockReward == 0 {
```

Comment:
When I run the benchmark program, block rewards exited here. We need to setup the block reward settings in [SetMockConfig](https://github.com/0chain/0chain/blob/staging/code/go/0chain.net/smartcontract/storagesc/benchmark_setup.go#L642)